flink-clients/src/test/java/org/apache/flink/client/program/PackagedProgramRetrieverImplTest.java

```diff
@@ -590,4 +567,25 @@ private JobGraph retrieveJobGraph(
                         packagedProgram, configuration, defaultParallelism, false);
         return PipelineExecutorUtils.getJobGraph(pipeline, configuration);
     }
+
+    private static List<String> extractRelativizedURLsForJarsFromDirectory(File directory)
+            throws MalformedURLException {
+        Preconditions.checkArgument(
+                directory.listFiles() != null,
+                "The passed File does not seem to be a directory: " + directory.getAbsolutePath());
```

Comment:
I initially thought the same. But objected it because in the end we're using `listFiles` in the code. And `listFiles` does not only return `null` in case of it not being a directory. But I udpated the error message to reflect that.